service/frontend/accessControlledHandler.go

```diff
@@ -724,10 +780,34 @@ func (a *AccessControlledWorkflowHandler) UpdateDomain(
 func (a *AccessControlledWorkflowHandler) isAuthorized(
 	ctx context.Context,
 	attr *authorization.Attributes,
+	scope metrics.Scope,
 ) (bool, error) {
+	sw := scope.StartTimer(metrics.CadenceAuthorizationLatency)
+	defer sw.Stop()
+
 	result, err := a.authorizer.Authorize(ctx, attr)
 	if err != nil {
+		scope.IncCounter(metrics.CadenceErrAuthorizeFailedCounter)
 		return false, err
 	}
-	return result.Decision == authorization.DecisionAllow, nil
+	isAuth := result.Decision == authorization.DecisionAllow
+	if !isAuth {
+		scope.IncCounter(metrics.CadenceErrUnauthorizedCounter)
+	}
+	return isAuth, nil
+}
+
+// getMetricsScopeWithDomain return metrics scope with domain tag
+func (a *AccessControlledWorkflowHandler) getMetricsScopeWithDomain(
```

Comment:
How about just making the following method a utility method that takes in a metrics client instead of duplicating the code 

```
// startRequestProfileWithDomain initiates recording of request metrics and returns a domain tagged scope
func (wh *WorkflowHandler) startRequestProfileWithDomain(scope int, d domainGetter) (metrics.Scope, metrics.Stopwatch) {
	var metricsScope metrics.Scope
	if d != nil {
		metricsScope = wh.GetMetricsClient().Scope(scope).Tagged(metrics.DomainTag(d.GetDomain()))
	} else {
		metricsScope = wh.GetMetricsClient().Scope(scope).Tagged(metrics.DomainUnknownTag())
	}
	sw := metricsScope.StartTimer(metrics.CadenceLatency)
	metricsScope.IncCounter(metrics.CadenceRequests)
	return metricsScope, sw
}
```

Reply:
updated.